plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -166,20 +174,35 @@ func PrioritizeNodes(pod *api.Pod, podLister algorithm.PodLister, priorityConfig
 		priorityFunc := priorityConfig.Function
 		prioritizedList, err := priorityFunc(pod, podLister, minionLister)
 		if err != nil {
-			return algorithm.HostPriorityList{}, err
+			return schedulerapi.HostPriorityList{}, err
 		}
 		for _, hostEntry := range prioritizedList {
 			combinedScores[hostEntry.Host] += hostEntry.Score * weight
 		}
 	}
+	if extender != nil && minionLister != nil {
+		nodes, err := minionLister.List()
+		if err != nil {
+			return schedulerapi.HostPriorityList{}, err
+		}
+
+		prioritizedList, err := extender.Prioritize(pod, &nodes)
```

Comment:
I left out the pods intentionally from the API, because I felt any inter pod dependencies that affect scheduling should reside in the kube scheduler (ex: Anti affinity scheduling). Any extender that would like to have the pods can get it from apiserver too. But its open for debate, let me know what your preference is.

Regarding the set of nodes passed to extender, its the same set as priorityFunc.  PrioritizeNodes gets a fake minion lister with the filtered nodes.